devito/ir/clusters/analysis.py

```diff
@@ -112,6 +128,7 @@ def _callback(self, clusters, d, prefix):
 
             test1 = len(prev) > 0 and any(dep.is_carried(i) for i in prev)
             if test1:
+                is_parallel_indep &= (dep.distance_mapper.get(d.root) == 0)
```

Comment:
`is_parallel_indep` is initialize as true  and never modified after that so
`is_parallel_indep = (dep.distance_mapper.get(d.root) == 0)` does the same thing without the need for line 119

Reply:
no. Always initialize variables in the scope as where they are accessed (read) for the first time. `is_parallel_indep` is read at line 142, which is the same scope as the initialization at line 119. If  you don't execute line 131, which is possible, you may easily get the most classic of the errors
```
NameError: name 'is_parallel_indep' is not defined
```

Reply:
no, one should always initialize variables in the same scope where they are accessed (read) for the first time.

`is_parallel_indep` is read at line 142, which is the same scope as that of the initialization at line 119. If  you don't execute line 131, which may be the case when you don't enter the `if`, you'll get the most classic of the errors
```
NameError: name 'is_parallel_indep' is not defined
```